ext/nnstreamer/tensor_filter/tensor_filter_executorch_llama.cc

```diff
@@ -0,0 +1,250 @@
+  ~executorch_llama_subplugin (){};
+
+  tensor_filter_subplugin &getEmptyInstance ();
+  void configure_instance (const GstTensorFilterProperties *prop);
+  void invoke (const GstTensorMemory *input, GstTensorMemory *output);
+  void invoke_dynamic (GstTensorFilterProperties *prop,
+      const GstTensorMemory *input, GstTensorMemory *output) override;
+  void getFrameworkInfo (GstTensorFilterFrameworkInfo &info);
+  int getModelInfo (model_info_ops ops, GstTensorsInfo &in_info, GstTensorsInfo &out_info);
+  int eventHandler (event_ops ops, GstTensorFilterFrameworkEventData &data);
+};
+
+const char *executorch_llama_subplugin::fw_name = "executorch-llama";
+
+/**
+ * @brief Method to get empty object.
+ */
+tensor_filter_subplugin &
+executorch_llama_subplugin::getEmptyInstance ()
+{
+  return *(new executorch_llama_subplugin ());
+}
+
+/**
+ * @brief Method to prepare/configure ExecuTorch instance.
+ */
+void
+executorch_llama_subplugin::configure_instance (const GstTensorFilterProperties *prop)
+{
+  try {
+    /* Load model file (.pte) */
+    if (!g_file_test (prop->model_files[0], G_FILE_TEST_IS_REGULAR)) {
+      const std::string err_msg
+          = "Given file " + (std::string) prop->model_files[0] + " is not valid";
+      throw std::invalid_argument (err_msg);
+    }
+
+    /* Load tokenizer */
+    if (!g_file_test (prop->model_files[1], G_FILE_TEST_IS_REGULAR)) {
+      const std::string err_msg
+          = "Given file " + (std::string) prop->model_files[1] + " is not valid";
+      throw std::invalid_argument (err_msg);
+    }
+
+    model_path = prop->model_files[0];
+    tokenizer_path = prop->model_files[1];
+
+    /* set llama runner */
+    runner = std::make_unique<example::Runner> (model_path, tokenizer_path, 0.8f);
+  } catch (const std::exception &e) {
+    /* throw exception upward */
+    throw;
+  }
+}
+
+/**
+ * @brief Method to execute the model.
+ */
+void
+executorch_llama_subplugin::invoke (const GstTensorMemory *input, GstTensorMemory *output)
+{
+  UNUSED (input);
+  UNUSED (output);
+
+  throw std::runtime_error (
+      "executorch-llama only supports invoke-dynamic mode. Set `invoke-dynamic=true`");
+}
+
+/**
+ * @brief Method to execute the model.
+ */
+void
+executorch_llama_subplugin::invoke_dynamic (GstTensorFilterProperties *prop,
+    const GstTensorMemory *input, GstTensorMemory *output)
+{
+  if (!input)
+    throw std::runtime_error ("Invalid input buffer, it is NULL.");
+  if (!output)
+    throw std::runtime_error ("Invalid output buffer, it is NULL.");
+
+  std::string prompt ((char *) input[0].data, input[0].size);
+  std::string result = "";
+
+  ET_CHECK_MSG (
+      runner->generate (prompt, 128,
+          [&result] (const std::string &generated) { result.append (generated); })
+          == Error::Ok,
+      "Generating text failed");
+
+  output[0].size = result.size ();
+  output[0].data = g_malloc0 (output[0].size);
+  memcpy (output[0].data, result.c_str (), result.size ());
```

Comment:
In my personal opinion,
It is necessary to check if the value of ```result.size()``` is valid and if ```output[0].data``` is allocated memory.
After that, doing ```memcpy``` seems to prevent the crash. how about think of this? right?

Reply:
`result` is valid std::string and g_malloc aborts if mem allocation failed. I think it is unnecessary to check here.